aptos-move/move-examples/nft_auction_house/sources/auction_house_own_listing.move

```diff
@@ -0,0 +1,531 @@
+// this example shows an auction house that stores user listing under auction house's account.
+
+module auction_house::auction_house_own_listing {
+
+    use aptos_framework::account;
+    use aptos_framework::coin;
+    use aptos_framework::event::{Self, EventHandle};
+    use aptos_framework::timestamp;
+    use aptos_std::guid::{Self, ID};
+    use aptos_std::table::{Self, Table};
+    use aptos_token::token::{Self, TokenId};
+    use auction_house::bid;
+    use auction_house::listing::{Self, Listing, ListingEvent};
+    use std::error;
+    use std::signer;
+    use std::string::String;
+    use std::vector;
+
+    //
+    // Constants
+    //
+
+    const AUCTION_DELAY_TIME_SEC: u64 = 300;
+
+    //
+    // Errors
+    //
+
+    /// Expiration time is invalid
+    const EINVALID_EXPIRATION_TIME: u64 = 1;
+
+    /// Start time is invalid
+    const EINVALID_START_TIME: u64 = 2;
+
+    /// Auction doesn't exist
+    const EAUCTION_NOT_EXIST: u64 = 3;
+
+    /// Bid increase less than minimal incremental
+    const EBID_INCREASE_TOO_SMALL: u64 = 4;
+
+    /// Auction ended
+    const EAUCTION_ENDED: u64 = 5;
+
+    /// Minimal incremental should be bigger than 0
+    const EBID_MIN_INCREMENTAL_IS_ZERO: u64 = 6;
+
+    /// Bid not found
+    const EBID_NOT_FOUND_FOR_AUCTION: u64 = 7;
+
+    /// Reserved operation for auction house owner
+    const EONLY_AUCTION_HOUSE_OWNER_CAN_PERFORM_THIS_OPERATION: u64 = 8;
+
+    /// Auction not ended
+    const EAUCTION_NOT_ENDED: u64 = 9;
+
+    struct AuctionHouseConfig has key {
+        market_fee_numerator: u64,
+        market_fee_denominator: u64,
+        fee_address: address,
+    }
+
+    struct Auctions<phantom CoinType> has key {
+        cur_auction_id: u64, // this is used to generate next auction_id
+        all_active_auctions: Table<u64, Auction<CoinType>>,
+        listing_event: EventHandle<ListingEvent>,
+        bid_event: EventHandle<BidEvent>,
+        cancel_bid_events: EventHandle<CancelBidEvent>
+    }
+
+    struct BidEvent has copy, drop, store {
+        market_address: address,
+        bid_id: ID,
+    }
+
+    struct CancelBidEvent has copy, drop, store {
+        market_address: address,
+        bid_id: ID,
+    }
+
+    struct Auction<phantom CoinType> has drop, store {
+        listing: Listing<CoinType>,
+        bids: vector<ID>,
+        highest_price: u64,
+        highest_bid: ID,
+        min_incremental_price: u64,
+    }
+
+    public entry fun initialize_auction_house(
+        account: &signer,
+        market_fee_numerator: u64,
+        market_fee_denominator: u64,
+        fee_address: address,
+    ) {
+        move_to(
+            account,
+            AuctionHouseConfig {
+                market_fee_denominator,
+                market_fee_numerator,
+                fee_address,
+            }
+        );
+    }
+
+    public entry fun initialize_auction<CoinType>(account: &signer) {
```

Comment:
Can this be combined with initialize_auction_house?

Reply:
no, we don't want to call the initialize_auction_house for each new coinType